### PR TITLE
adds build files for review

### PR DESCRIPTION
**Pull request by pbcornell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlurton** [comment]
Is there a reason you're on Gradle 4.1? I don't see a reason for an new code base to be using such an old version of Gradle.

I'd recommend upgrading to 5.2.1.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,107 @@
+buildscript {
+  ext.kotlin_version = "[1.2,)"
+
+  dependencies {
+    repositories {
+      mavenCentral()
+      jcenter()
```
Comment (dlurton) [inline build.gradle]: This might be obvious but I'm not certain why you need both `mavenCentral` and `jcenter` here?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    classpath "org.jetbrains.dokka:dokka-gradle-plugin:0.9.17"
```
Comment (dlurton) [inline build.gradle]: I think this won't be needed if you update to gradle 5.2.1 and use the more recent plugin syntax I mention below.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
+}
+sourceSets {
+  main.java.srcDirs = ["src"]
+  main.kotlin.srcDirs = ["src"]
+  test.java.srcDirs = ["test"]
+  test.kotlin.srcDirs = ["test"]
+}
+
+checkstyle {
```
Comment (dlurton) [inline build.gradle]: Not sure this is doing anything since there is no java here.

File: build.gradle (hunk 1 above)
Comment (pbcornell) [reply build.gradle]: `jcenter` is required to resolve the `dokka` plugin

File: build.gradle (hunk 2 above)
Comment (pbcornell) [reply build.gradle]: correct, removed

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      mavenCentral()
+      jcenter()
+    }
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    classpath "org.jetbrains.dokka:dokka-gradle-plugin:0.9.17"
+  }
+}
+
+apply plugin: "kotlin"
+apply plugin: "checkstyle"
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: removed

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+apply plugin: "org.jetbrains.dokka"
+apply plugin: "maven-publish"
+apply plugin: "signing"
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: switched to new `plugins` syntax

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+group = "software.amazon.ion"
+version = "0.8.0-SNAPSHOT"
+
+repositories { mavenCentral() }
+
+dependencies {
+  compile "software.amazon.ion:ion-java:[1.2,)"
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: no particular need to upgrade ion-java, but no reason not to either;  updated to 1.3

also moved away from old `compile` dependency type

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: agreed

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: agreed

File: build.gradle (hunk 3 above)
Comment (pbcornell) [reply build.gradle]: removed

File: settings.gradle [hunk 9]
```diff
@@ -0,0 +1,2 @@
+enableFeaturePreview('STABLE_PUBLISHING')
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply settings.gradle]: correct, removed

File: build.gradle [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+}
+sourceSets {
+  main.java.srcDirs = ["src"]
+  main.kotlin.srcDirs = ["src"]
+  test.java.srcDirs = ["test"]
+  test.kotlin.srcDirs = ["test"]
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply build.gradle]: `main.java.srcDirs` and `test.java.srcDirs` aren't needed, removed

Not sure i agree with doing things "the normal way" for this repository.  The repo is called ion-schema-_kotlin_ -- why does it make sense to further nest the code?  I'd be happy to make this type of change if the repo warranted it, but I don't see much (if any) value in doing so at this time.

**Comment by dlurton** [comment]
Looking at why the gradle build seems to be failing in travis, I see that [`gradle` is being executed instead of `./gradlew`](https://travis-ci.org/amzn/ion-schema-kotlin/jobs/517340896#L462)--I wonder if the version of gradle installed in Travis is not 5.3.1?  This is odd because the [travis documentation](https://docs.travis-ci.com/user/languages/java/#projects-using-gradle) for says that if a `gradlew` exists then it will be used.  Theoretically, this should ensure the same version of gradle is used in Travis...

**Pull request merged**